server/rest/large_image.py

```diff
@@ -184,6 +196,8 @@ def countThumbnails(self, params):
             query = {'isLargeImageThumbnail': True, 'attachedToType': 'item'}
             if entry is not None:
                 query['thumbnailKey'] = entry
+            elif associatedImages:
+                query['thumbnailKey'] = {'$regex': '"imageKey":'}
```

Comment:
What's the rationale for storing this as JSON and using a regex query instead of storing it as BSON in the database?

Reply:
As long as we are OK with the fact that these endpoints will require a full collection scan.

Reply:
In mongo `db.file.find({thumbnailKey: {width: 160, height: 100}}).count()` gets a different answer than `db.file.find({thumbnailKey: {height: 100, width: 160}}).count()`.  By using json with a sorted string, this avoids the dictionary of image properties being in a indeterminate order.  We could have used an ordered dictionary instead, which would make it easier to do mongo subqueries.

Reply:
Won't the following query work?

```json
{
    "thumbnailKey.width": 160,
    "thumbnailKey.height": 100
}
``` 

If so, that would allow for an indexed lookup

Reply:
That query will work, but doesn't exclude documents with other properties (that is, it would match `thumbnailKey: {width: 160, height: 100, encoding: "PNG"}`).  Using a mongo query of `{thumbnailKey: {width: 160, height: 100}}` requires all subproperties to match and there to be no additional subproperties but also requires them to be **in order**.

Reply:
That's really odd, I didn't know that.

Reply:
Switching to used ordered dicts instead of sorted json would be easy, and probably reduce future confusion.  If you think it is worth it, I can make a PR doing so.

Reply:
It's your call, I'm just surprised about that behavior in mongodb.